docs/source/development-guide/release-workflow.rst

```diff
@@ -27,16 +37,17 @@ Nominal releases
 #. Make sure the ``dev`` branch is up-to-date with any changes you want included in the release (i.e. merge in any
    feature branches using the nominal :ref:`git & GitHub Workflow <git-and-github-workflow>`).
 #. Create a new version branch off of ``dev``.  The name of the branch should match the version number to be used for
-   the release, which should follow the :ref:`software versioning <versioning>` conventions. The patch number should be
-   marked with a ``x``, and the name should be prepended with ``v`` (e.g. ``v0.1.x``).
+   the release, which should follow the :ref:`software versioning <versioning>` conventions. The name should be
+   prepended with ``v`` (e.g. ``v0.1.0``).
 #. Make any release-specific commits to the new version branch using the nominal ``git add``/``git commit`` cycle. This
    may include commits that add release notes, or update version numbers in various configurations.
 #. Push the version branch to the main ``IMAP-Science-Operations-Center`` ``imap_processing`` repo (i.e. ``upstream``).
 #. Create a `new release <https://github.com/IMAP-Science-Operations-Center/imap_processing/releases>`_, using the
    version branch as the ``Target`` branch. Assign a new tag with the specific version number, including the patch
    number (e.g. ``v0.1.0``).
-#. In GitHub, create a pull request that merges the version branch into ``dev``. Proceed with the nominal review & merge
-   process described in steps (10) and (11) in the :ref:`git & GitHub Workflow <git-and-github-workflow>` section.
+#. If there have been release-specific commits, In GitHub, create a pull request that merges the version branch into
+   ``dev``. Proceed with the nominal review & merge process described in steps (10) and (11) in the :ref:`git & GitHub
+   Workflow <git-and-github-workflow>` section.
```

Comment:
I am not good with release. I tried to release last month to fix a bug and I think I had to merge PR before I create a `new release`. I tried to see if I can find notes but I couldn't. Is that what you do also?

Reply:
Good point, I think you are right about the ordering. That is how I do it for `sds_data_manager`, which usually requires some release-specific commits to update version numbers. 

I updated the docs to reflect the correct order, and also added some more notes.